ql/src/test/org/apache/hadoop/hive/ql/io/parquet/serde/TestParquetTimestampUtils.java

```diff
@@ -389,6 +390,26 @@ public void testIllegalInt64TimestampStrings() {
     verifyInt64TimestampValue("2262-04-11 23:47:16.854775808", LogicalTypeAnnotation.TimeUnit.NANOS, false);
   }
 
+  @Test
+  public void testTimestamp9999() {
```

Comment:
What do you think of refactoring and moving the test in `TestParquetTimestampsHive2Compatibility` which exactly about compatibility with Hive 2?

Reply:
I think for a single tests we should not create a separate test class.
Added comment to make clear what we are testing here.

Reply:
I was thinking to move it inside the existing class, not create a new one.

Reply:
Or rather enrich the tests inside `TestParquetTimestampsHive2Compatibility` to account for this edge case.

Reply:
Th ave never seen `TestParquetTimestampsHive2Compatibility`, so it is good that you highlighted here. OTOH that is a parameterized test, without TZ, so I think it would be a full rewrite to get this work for this specific TZ and TS

Reply:
How about something like the following:

```java
  private static Stream<String> generateTimestamps() {
    return Stream.concat(Stream.of("9999-12-31 23:59:59.999"), Stream.generate(new Supplier<String>() {
```

Reply:
`TestParquetTimestampsHive2Compatibility` also uses multiple timezones so I think it should be fine and adding another one should be trivial.